frameworks/Rust/iron/src/main.rs

```diff
@@ -1,36 +1,226 @@
 extern crate iron;
-extern crate router;
+extern crate persistent;
+#[macro_use] extern crate router;
+extern crate serde;
+extern crate serde_json;
+extern crate hyper;
+extern crate rand;
+extern crate r2d2;
+extern crate postgres;
+extern crate r2d2_postgres;
+extern crate mustache;
 extern crate rustc_serialize;
 
-use iron::{Iron, Request, Response, IronResult};
+use iron::prelude::*;
 use iron::status;
-use router::Router;
-use rustc_serialize::json;
-use iron::mime::Mime;
-use iron::headers::Server;
 use iron::modifiers::Header;
+use iron::typemap::Key;
+use hyper::header::{Server, ContentType};
+use rand::distributions::{Range, IndependentSample};
+use r2d2_postgres::{PostgresConnectionManager, TlsMode};
+use persistent::{Read};
+use r2d2::Pool;
 
-#[derive(RustcDecodable, RustcEncodable)]
-struct Message {
-    message: String,
+include!(concat!(env!("OUT_DIR"),"/main_types.rs"));
+
+pub type PostgresPool = Pool<PostgresConnectionManager>;
+
+struct DbPool;
+impl Key for DbPool { type Value = PostgresPool; }
+
+struct FortuneTemplate;
+impl Key for FortuneTemplate { type Value = mustache::Template; }
+
+#[derive(RustcEncodable)]
+struct FortuneRow {
+    id: i32,
+    message: String
 }
 
 fn main() {
-    let mut router = Router::new();
-    router.get("/json", json_handler);
-    router.get("/plaintext", plaintext_handler);
-
-    Iron::new(router).http("0.0.0.0:8080").unwrap();
+    let r2d2_config = r2d2::Config::default();
+    let pg_conn_manager = PostgresConnectionManager::new(
+        "postgres://benchmarkdbuser:benchmarkdbpass@localhost/hello_world",
```

Comment:
Could you add a line in your `setup.sh` file that replaces "localhost" here with the current value of the environment variable $DBHOST? Rust/Iron will pass on Travis and on your local instance because they are running on a single machine, but our production setups use separate machines for the app server and the database server, so this connection will fail in that case. Here is an [example](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Elixir/phoenix/setup.sh#L5) of what your `setup.sh` line should look like.